### PR TITLE
Feature/third deliverable

### PR DESCRIPTION
**Pull request by majezanu**
Add a new endpoint
The endpoint must read items from the CSV concurrently using a worker pool
The endpoint must support the following query params:
type: Only support "odd" or "even"
items: Is an Int and is the number of valid items you need to display as a response
items_per_workers: Is an Int and is the number of valid items the worker should append to the response
Reject the values according to the query param  (you could use an ID column)
Instruct the workers to shut down according to the query param  collected
The result should be displayed as a response
The response should be displayed when:
The workers reached the limit
EOF
Valid items completed

### CODE REVIEW

File: internal/implementations/controller/pokemon_controller.go [hunk 1]
```diff
@@ -18,6 +18,18 @@ func responseError(c controller.Context, err error) error {
 	return c.JSON(errorResponse.Code, errorResponse)
 }
 
+func (p pokemonController) GetMultiple(c controller.Context) error {
+	var payload model.MultipleFilter
+	if err := c.Bind(&payload); err != nil {
+		return responseError(c, err)
+	}
+	pokemonList, err := p.pokemonInteractor.GetMultiple(payload.IdType, payload.Items, payload.ItemsPerWorker)
```
Comment (juanpabs) [inline internal/implementations/controller/pokemon_controller.go:26]: cuando envio solo uno de los query parameters siempre me responde null este endpoint. solo enviando los 3 funciona

File: internal/implementations/interactor/pokemon_interactor.go [hunk 2]
```diff
@@ -13,10 +13,24 @@ type pokemonUseCase struct {
 	Client     client.PokemonClient
 }
 
+const EVEN = "even"
```
Comment (juanpabs) [inline internal/implementations/interactor/pokemon_interactor.go:16]: excellent! you use constants!!

File: internal/implementations/repository/pokemon_repository.go [hunk 3]
```diff
@@ -113,6 +115,102 @@ func (p pokemonRepository) FindAll() (pokemonList []model.Pokemon, err error) {
...
+			break
+		}
+		result = append(result, res)
+	}
+
+	return
+}
+
 func (p pokemonRepository) Save(pokemon *model.Pokemon) (err error) {
 	file, err := p.OpenerCloser.OpenToWrite()
```
Comment (juanpabs) [inline internal/implementations/repository/pokemon_repository.go:215]: falta manejar este error

Comment (juanpabs) [reply internal/implementations/repository/pokemon_repository.go:215]: En general super buen trabajo :D

Comment (majezanu) [reply internal/implementations/repository/pokemon_repository.go:215]: Oh, muy bien. Muchas gracias. Iré corrigiendo las cosas.

**Pull request merged**